ledger/cow.go

```diff
@@ -125,18 +133,28 @@ func (cb *roundCowState) txnCounter() uint64 {
 	return cb.lookupParent.txnCounter() + uint64(len(cb.mods.Txids))
 }
 
-func (cb *roundCowState) put(addr basics.Address, old basics.AccountData, new basics.AccountData) {
+func (cb *roundCowState) put(addr basics.Address, old basics.AccountData, new basics.AccountData, newCreatables []basics.CreatableLocator, deletedCreatables []basics.CreatableLocator) {
 	prev, present := cb.mods.accts[addr]
 	if present {
 		cb.mods.accts[addr] = accountDelta{old: prev.old, new: new}
 	} else {
 		cb.mods.accts[addr] = accountDelta{old: old, new: new}
 	}
 
-	// Get which asset indices were created and deleted, and update state
-	assetDeltas := getChangedAssetIndices(addr, accountDelta{old: old, new: new})
-	for aidx, delta := range assetDeltas {
-		cb.mods.assets[aidx] = delta
+	for _, cl := range newCreatables {
+		cb.mods.creatables[cl.Index] = modifiedCreatable{
+			ctype:   cl.Type,
+			creator: addr,
```

Comment:
Nit: I don't think this ever happens, but the redundancy here assumes `addr` and `cl.Creator` are the same.

Reply:
yeah, this is a good point; resolved so that it uses the creatable's `Creator` field like it should.